client/shipping-label/state/actions.js

```diff
@@ -1,15 +1,99 @@
-import * as FormActions from 'settings/state/actions';
+import saveForm from 'lib/save-form';
+import noop from 'lodash/noop';
+import fill from 'lodash/fill';
+import flatten from 'lodash/flatten';
+import omit from 'lodash/omit';
+import moment from 'moment';
+import printDocument from 'lib/utils/print-document';
+import * as NoticeActions from 'state/notices/actions';
 
 export const OPEN_PRINTING_FLOW = 'OPEN_PRINTING_FLOW';
 export const EXIT_PRINTING_FLOW = 'EXIT_PRINTING_FLOW';
+export const UPDATE_ADDRESS_VALUE = 'UPDATE_ADDRESS_VALUE';
+export const UPDATE_PACKAGE_WEIGHT = 'UPDATE_PACKAGE_WEIGHT';
+export const UPDATE_RATE = 'UPDATE_RATE';
+export const UPDATE_PAPER_SIZE = 'UPDATE_PAPER_SIZE';
+export const PURCHASE_LABEL_REQUEST = 'PURCHASE_LABEL_REQUEST';
+export const PURCHASE_LABEL_RESPONSE = 'PURCHASE_LABEL_RESPONSE';
 
-export const openPrintingFlow = () => ( dispatch ) => {
-	dispatch( { type: OPEN_PRINTING_FLOW } );
-	dispatch( FormActions.setFormProperty( 'pristine', false ) );
-	dispatch( FormActions.nextStep() );
+export const openPrintingFlow = () => {
+	return { type: OPEN_PRINTING_FLOW };
 };
 
-export const exitPrintingFlow = () => ( dispatch ) => {
-	dispatch( { type: EXIT_PRINTING_FLOW } );
-	dispatch( FormActions.resetFlow() );
+export const exitPrintingFlow = () => {
+	return { type: EXIT_PRINTING_FLOW };
+};
+
+export const updateAddressValue = ( group, name, value ) => {
+	return {
+		type: UPDATE_ADDRESS_VALUE,
+		group,
+		name,
+		value,
+	};
+};
+
+export const updatePackageWeight = ( packageIndex, value ) => {
+	return {
+		type: UPDATE_PACKAGE_WEIGHT,
+		packageIndex,
+		value,
+	};
+};
+
+export const updateRate = ( packageIndex, value ) => {
+	return {
+		type: UPDATE_RATE,
+		packageIndex,
+		value,
+	};
+};
+
+export const updatePaperSize = ( value ) => {
+	return {
+		type: UPDATE_PAPER_SIZE,
+		value,
+	};
+};
+
+export const purchaseLabel = () => ( dispatch, getState, { callbackURL, nonce, submitMethod } ) => {
+	let error = null;
+	let response = null;
+	const setError = ( err ) => error = err;
+	const setSuccess = ( success, json ) => {
+		if ( success ) {
+			response = json;
+		}
+	};
+	const setIsSaving = ( saving ) => {
+		if ( saving ) {
+			dispatch( { type: PURCHASE_LABEL_REQUEST } );
+		} else {
+			dispatch( { type: PURCHASE_LABEL_RESPONSE, response, error } );
+			if ( error ) {
+				dispatch( NoticeActions.errorNotice( error.toString(), {
+					duration: 7000,
+				} ) );
+			} else {
+				printDocument( 'data:application/pdf;base64,' + response[ 0 ].image ); // TODO: Figure out how to print multiple PDFs
+			}
+		}
+	};
+
+	const form = getState().shippingLabel.form;
+	const formData = {
+		origin: form.origin.values,
+		destination: form.destination.values,
+		label_size: form.preview.values.paper_size,
+		ship_date: moment().add( 1, 'days' ).format( 'YYYY-MM-DD' ),
```

Comment:
For the record ( @kellychoffman ) EasyPost doesn't require ShipDate, so I'm removing this and you don't have to include in the UI (at least until/if we integrate with USPS directly)